api/orders.js

```diff
@@ -0,0 +1,100 @@
+const express = require("express");
+const router = express.Router();
+const {
+  getAllOrders,
+  getOrderById,
+  createOrder,
+  updateOrder,
+  addProductToOrder,
+  destroyOrder,
+} = require("../db");
+const { requireUser } = require("./utils");
+
+router.get("/", async (req, res, next) => {
+  const orders = await getAllOrders();
+
+  res.send(orders);
+});
+
+router.post("/", requireUser, async (req, res) => {
+  const { subtotal } = req.body;
+  const orderData = {};
+
+  try {
+    orderData.creatorId = req.user.id;
+    orderData.subtotal = subtotal;
+
+    const order = await createOrder(orderData);
+
+    if (!order) {
+      next({
+        name: "UserNotAuthorized",
+        message: "User is not authorized to create an order.",
+      });
+    }
+
+    res.send(order);
+  } catch ({ name, message }) {
+    next({ name, message });
+  }
+});
+
+router.patch("/:orderId", async (req, res, next) => {
+  const { orderId } = req.params;
+  const { subtotal } = req.body;
+
+  try {
+    const orderById = await getOrderById(orderId);
+
+    if (orderById.creatorId === req.user.id) {
+      const updatedOrder = await updateOrder({
+        id: orderId,
+        subtotal,
+      });
+      res.send(updatedOrder);
+    } else {
+      next({
+        name: "userUnauthorizeToUpdate",
+        message: "User is not authorized to update this order.",
+      });
+    }
```

Comment:
if (!orderById.creatorId) {
// guest order
} else if (orderById.creatorId === req.user.id) {
// user order
}

Reply:
Done, but check it before we resolve the conversation.